ports/geckolib/gecko_bindings/sugar/refptr.rs

```diff
@@ -0,0 +1,91 @@
+use std::marker::PhantomData;
+use std::mem::{forget, transmute};
+use std::ptr::{read, null};
+use std::sync::Arc;
+
+/// Helper trait for conversions between FFI Strong/Borrowed types and Arcs
+pub unsafe trait RefCounted where Self: Sized {
+    /// Gecko's name for the type
+    /// This is equivalent to ArcBox<Self>
+    type FFIType: Sized;
+    fn with<F, Output>(raw: Borrowed<Self::FFIType>, cb: F) -> Output
+               where F: FnOnce(&Arc<Self>) -> Output {
+        debug_assert!(!raw.is_null());
+
+        let owned = unsafe { Self::borrowed_into(raw) };
+        let result = cb(&owned);
+        forget(owned);
```

Comment:
I know that gecko is going to build with panic=abort, but this still triggers my "definitely not unwind safe" senses. I'm pretty sure that servo is built with unwinding panics, so if this code ever made its way into servo proper, it would be bad. Can we add some sort of comment to note that this isn't unwind safe?